### PR TITLE
Readme improvements

### PR DESCRIPTION
**Pull request by DanHickstein**
This is basically just https://github.com/PyAbel/PyAbel/pull/111, but re-done to have fewer silly commits and also to incorporate @stggh's changes that were refusing to appear in my branch. 

In addition, I changed `CONTRIBUTING.md` to an `.rst` file since this will make it easier to incorporate it into the sphinx documentation on readthedocs.org.

You can see how the new readme and contributing documents look here:
https://github.com/DanHickstein/PyAbel/tree/readme_improvements
https://github.com/DanHickstein/PyAbel/blob/readme_improvements/CONTRIBUTING.rst

And here is the readme and contributing on readthedocs:
http://pyabel-dh-fork.readthedocs.org/en/latest/readme_link.html
http://pyabel-dh-fork.readthedocs.org/en/latest/contributing_link.html

Also, the examples now appear in their own section: 
http://pyabel-dh-fork.readthedocs.org/en/latest/examples.html

@PyAbel/developers, please proofread and let me know if you would like to see changes. Thanks!

Sidenote: I also did: `chmod 644 abel/tests/test_*.py` in order to make all of the tests not executable. `nosetests` doesn't run executable files, so we need to keep them not executable (except `run.py`) in order for the test to work.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+
+Documentation
+-------------
+
+PyAbel uses Sphinx and `Napoleon <http://sphinxcontrib-napoleon.readthedocs.org/en/latest/index.html>`_ to process Numpy style docstrings, and is synchronized to `pyabel.readthedocs.org <http://pyabel.readthedocs.org>`_. To build the documentation locally, you will need Sphinx, the `recommonmark <https://github.com/rtfd/recommonmark>`_ package, and the `sphinx_rtd_theme <https://github.com/snide/sphinx_rtd_theme/>`_. You can install all this this using ::
+
+	pip install sphinx
+	pip install recommonmark
+	pip isnstall sphinx_rtd_theme
```
Comment (DhrubajyotiDas) [inline CONTRIBUTING.rst]: Typo here in spelling 'install'

**Comment by DanHickstein** [comment]
Can anyone figure out what is going wrong with the 64-bit Appveyor build? Something to do with the Cython module. Maybe @rth can figure it out...

**Comment by rth** [comment]
Merging this as IMO, the issue with AppVeyor is not introduced by this PR, but came from some minor upgrade in Conda of scipy/numpy/mkl versions. I had experiences similar issues in the past (applications failing to run on Windows with slightly different versions of scipy/numpy/mkl  without apparent resons). I will make another PR to fix this build. If cython related  issues become too time consuming, we can also consider disabling the Cython build on Windows, as at present it does not have a critical role in PyAbel.

**Comment by DanHickstein** [comment]
Thanks for having a look @rth!

**Pull request merged**